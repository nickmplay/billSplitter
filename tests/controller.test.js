const { Bill, Person } = require('../src/bill');
const { ViewBill } = require('../src/view');
const { billController } = require('../src/controller');

// const newView = new ViewBill();
// const newModel = new Bill(100);
// const fakeId = 1462361249717;
// //same as mock ids for snapshots
// newModel.people[0].id = fakeId;
// newModel.people[1].id = fakeId + 1;

// const dom = document.createElement("div");
// const testCon = new billController(newView, newModel, dom);
// testCon.createApp();

//implementing before each testing structure
let newView, newModel, fakeId, dom, testCon;
const startingNumber = 2;

beforeEach(()=>{
  newView = new ViewBill();
  newModel = new Bill(100);
  fakeId = 1462361249717;
  //same as mock ids for snapshots
  newModel.people[0].id = fakeId;
  newModel.people[1].id = fakeId + 1;

  dom = document.createElement("div");
  testCon = new billController(newView, newModel, dom);
  testCon.createApp();
});

test('instantiate a controller class', () => {
  expect( testCon.model.countPeople() ).toBe(startingNumber);
  expect(dom).toMatchSnapshot();
});

test('add a person', ()=>{
  expect( testCon.model.countPeople() ).toBe(startingNumber);
  // dom.querySelectorAll("button")[2].click();
  dom.querySelector("button.addPersonBtn").click()
  expect( testCon.model.countPeople() ).toBe(3);
  expect( dom.querySelectorAll("li").length ).toBe(3);
  const shares = [];
  for(let i=0; i<3; i++){
    shares.push( dom.querySelectorAll("span")[i].innerHTML );
  }
  expect( shares.join("") ).toBe('Share: 33.33Share: 33.33Share: 33.33'); 
});

test('remove a person', ()=>{
  expect( testCon.model.countPeople() ).toBe(startingNumber);
  dom.querySelector("button").click();
  expect( testCon.model.countPeople() ).toBe(1);
  expect( dom.querySelectorAll("li").length ).toBe(1);
  expect( dom.querySelectorAll("span")[0].innerHTML ).toBe('Share: 100'); 
});