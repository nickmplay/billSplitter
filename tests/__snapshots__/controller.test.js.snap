// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit a person 1`] = `
<div>
  <div
    class="totalBill"
  >
    <label
      for="amountInp"
    >
      Enter total bill to split:
    </label>
    <input
      class="input-total"
      name="amountInp"
      placeholder="Enter amount"
      type="number"
      value="100"
    />
  </div>
  <ol>
    <li
      p-id="1462361249717"
    >
      To pay 
      <input
        <=""
        class="p-amount"
        input=""
        placeholder="0"
        type="number"
        value="10"
      />
      <select>
        <option
          value="split"
        >
          split
        </option>
        <option
          selected="true"
          value="more"
        >
          more
        </option>
        <option
          value="less"
        >
          less
        </option>
        <option
          value="fixed"
        >
          fixed
        </option>
      </select>
      <span
        p-id="1462361249717"
      >
        Share: 55
      </span>
      <button>
        X
      </button>
    </li>
    <li
      p-id="1462361249718"
    >
      To pay 
      <input
        <=""
        class="p-amount"
        disabled=""
        input=""
        placeholder="0"
        type="number"
      />
      <select>
        <option
          selected=""
          value="split"
        >
          split
        </option>
        <option
          value="more"
        >
          more
        </option>
        <option
          value="less"
        >
          less
        </option>
        <option
          value="fixed"
        >
          fixed
        </option>
      </select>
      <span
        p-id="1462361249718"
      >
        Share: 45
      </span>
      <button>
        X
      </button>
    </li>
  </ol>
  <button
    class="addPersonBtn"
  >
    Add Person
  </button>
</div>
`;

exports[`instantiate a controller class 1`] = `
<div>
  <div
    class="totalBill"
  >
    <label
      for="amountInp"
    >
      Enter total bill to split:
    </label>
    <input
      class="input-total"
      name="amountInp"
      placeholder="Enter amount"
      type="number"
      value="100"
    />
  </div>
  <ol>
    <li
      p-id="1462361249717"
    >
      To pay 
      <input
        <=""
        class="p-amount"
        disabled=""
        input=""
        placeholder="0"
        type="number"
      />
      <select>
        <option
          selected=""
          value="split"
        >
          split
        </option>
        <option
          value="more"
        >
          more
        </option>
        <option
          value="less"
        >
          less
        </option>
        <option
          value="fixed"
        >
          fixed
        </option>
      </select>
      <span
        p-id="1462361249717"
      >
        Share: 50
      </span>
      <button>
        X
      </button>
    </li>
    <li
      p-id="1462361249718"
    >
      To pay 
      <input
        <=""
        class="p-amount"
        disabled=""
        input=""
        placeholder="0"
        type="number"
      />
      <select>
        <option
          selected=""
          value="split"
        >
          split
        </option>
        <option
          value="more"
        >
          more
        </option>
        <option
          value="less"
        >
          less
        </option>
        <option
          value="fixed"
        >
          fixed
        </option>
      </select>
      <span
        p-id="1462361249718"
      >
        Share: 50
      </span>
      <button>
        X
      </button>
    </li>
  </ol>
  <button
    class="addPersonBtn"
  >
    Add Person
  </button>
</div>
`;
